File: Parth Mathur/H-Index.java
class Solution {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);
        int n=citations.length;
        int ans=-1;
        int l=0;
        int h=citations.length-1;
        while(l<=h){
            int mid=l+(h-l)/2;
            if(citations[mid]==n-mid) return citations[mid];
            else if(citations[mid]<n-mid) l=mid+1; 
            else h=mid-1; 
        }
        return n-l;
    }
}

File: Mudit Choraria/h-index.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n = citations.size();
        vector<int> count(n + 1);
        
        for(auto citation: citations) {
            ++count[min(citation, n)];
        }
        
        int acc = 0, ans = 0;
        for(int i = n; i >= 0; i--) {
            acc += count[i];
            if(acc >= i) {
                ans = i;
                break;
            }
        }
        return ans;
    }
};